api/src/models/utils/sort.ts

```diff
@@ -29,9 +32,15 @@ export const Sort = <
 	t
 		.Transform(
 			t.Array(
-				t.UnionEnum([
-					...values,
-					...values.map((x: T[number]) => `-${x}` as const),
+				t.Union([
+					t.UnionEnum([
+						...values,
+						...values.map((x: T[number]) => `-${x}` as const),
+					]),
+					t.Union([
+						t.TemplateLiteral("random:${number}"),
+						t.TemplateLiteral("-random:${number}"),
+					]),
 				]),
```

Comment:
since we support either sorts or random (but not both at the same time, the type should probably be:

```tsx
t.Union(
	t.Literal("random"),
	t.TemplateLiteral("random:${number}"),
	t.Array([
		t.UnionEnum([
			...values,
			...values.map((x: T[number]) => `-${x}` as const),
		]),
	]),
)
```